rafx-framework/src/resources/descriptor_sets/descriptor_set_pool_chunk.rs

```diff
@@ -17,6 +21,453 @@ use rafx_api::{
+            self.buffers,
+            &self.descriptor_set_layout.get_raw(),
+            self.descriptor_set_array,
+            descriptor_set_index as u32,
+            element_key,
+            element,
+        )?;
+
+        Ok(())
+    }
+
+    fn write_buffer_data<T: Copy>(
+        &mut self,
+        element_key: &DescriptorSetElementKey,
+        data: &[T],
+    ) -> RafxResult<()> {
+        let descriptor_set_index = self.descriptor_set_index as u32;
+
+        log::trace!(
+            "Process descriptor set write for {:?} {:?}. layout {:?}",
+            descriptor_set_index,
+            element_key,
+            self.descriptor_set_layout,
+        );
+
+        let buffer = self
+            .buffers
+            .buffer_sets
+            .get_mut(&DescriptorSetBindingKey {
+                dst_binding: element_key.dst_binding,
+            })
+            .unwrap();
+
+        if data.len() as u32 > buffer.buffer_info.per_descriptor_size {
+            panic!(
+                "Wrote {} bytes to a descriptor set buffer that holds {} bytes layout: {:?}",
+                data.len(),
+                buffer.buffer_info.per_descriptor_size,
+                self.descriptor_set_layout
+            );
+        }
+
+        if data.len() as u32 != buffer.buffer_info.per_descriptor_size {
+            log::warn!(
+                "Wrote {} bytes to a descriptor set buffer that holds {} bytes layout: {:?}",
+                data.len(),
+                buffer.buffer_info.per_descriptor_size,
+                self.descriptor_set_layout
+            );
+        }
+
+        let offset = buffer.buffer_info.per_descriptor_stride * descriptor_set_index;
+
+        log::trace!(
+            "Writing {} bytes to internal buffer to set {} at offset {}",
+            data.len(),
+            descriptor_set_index,
+            offset
+        );
+        buffer
+            .buffer
+            .copy_to_host_visible_buffer_with_offset(&data, offset as u64)
+            .unwrap();
+
+        Ok(())
+    }
+
+    pub fn set_image(
+        &mut self,
+        binding_index: u32,
+        image_view: &ResourceArc<ImageViewResource>,
+    ) -> RafxResult<()> {
+        self.set_image_array_element(
+            binding_index,
+            0,
+            DescriptorSetWriteElementImageValue::Resource(image_view.clone()),
+        )
+    }
+
+    pub fn set_images(
+        &mut self,
+        binding_index: u32,
+        image_views: &[Option<&ResourceArc<ImageViewResource>>],
+    ) -> RafxResult<()> {
+        for (index, image_view) in image_views.iter().enumerate() {
+            if let Some(image_view) = image_view.as_ref() {
+                self.set_image_array_element(
+                    binding_index,
+                    index,
+                    DescriptorSetWriteElementImageValue::Resource((*image_view).clone()),
+                )?
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn set_image_at_index(
```

Comment:
Can any of these multitude of methods be de-duplicated or made common with `DynDescriptorSet`? Maybe define as a `Trait` so that the IDE can track them together?